src/updater/updater.cpp

```diff
@@ -413,20 +439,128 @@ updateTurnPenalties(const UpdaterConfig &config,
 
         if (turn_weight_penalty < 0)
         {
-            util::Log(logWARNING) << "Negative turn penalty at " << turn_index.from_id << ", "
-                                  << turn_index.via_id << ", " << turn_index.to_id
+            util::Log(logWARNING) << "Negative turn penalty at " << internal_turn.from_id << ", "
+                                  << internal_turn.via_id << ", " << internal_turn.to_id
                                   << ": turn penalty " << turn_weight_penalty;
         }
     }
 
     return updated_turns;
 }
+
+bool IsRestrictionValid(const Timezoner &tz_handler,
+                        const extractor::TurnRestriction &turn,
+                        const std::vector<util::Coordinate> &coordinates,
+                        const extractor::PackedOSMIDs &osm_node_ids)
+{
+    // get restriction's lon/lat coords
+    const auto via_node = osm_node_ids[turn.via.node];
+    const auto from_node = osm_node_ids[turn.from.node];
+    const auto to_node = osm_node_ids[turn.to.node];
+    const auto &lon = static_cast<double>(toFloating(coordinates[turn.via.node].lon));
+    const auto &lat = static_cast<double>(toFloating(coordinates[turn.via.node].lat));
+    const auto &condition = turn.condition;
+
+    // Get local time of the restriction
+    const auto &local_time = tz_handler.GetLocalTime(point_t{lon, lat});
+
+    // TODO: check restriction type [:<transportation mode>][:<direction>]
+    // http://wiki.openstreetmap.org/wiki/Conditional_restrictions#Tagging
+
+    // TODO: parsing will fail for combined conditions, e.g. Sa-Su AND weight>7
+    // http://wiki.openstreetmap.org/wiki/Conditional_restrictions#Combined_conditions:_AND
+
+    if (condition.empty())
```

Comment:
Move this at the beginning of the function for an early exit. In that case we don't actually need to do any of the operations above.